flamingchickens1540/inflatedchickens
# WebSocket Queue

### Description
The queuing system's WebSocket server backend and a minimal frontend for testing.

### Features
- [x] Backend support for queuing, dequeuing, and match creation
- [x] Backend syncs the current match with all clients
- [x] Minimal frontend screen for queuing and dequeuing as a scout
- [x] Minimal Admin Dashboard for queuing matches

###  Testing
1. `bun install`
2. `bun run dev`
3. Open two pages, one for the `/admin` route and one for the `/` route
4. In the '/' route, click the `scout` button to enter the queue
5. Verify that you can leave and rejoin the queue
6. In the admin page, fill in all the inputs with whatever test strings you want
7. Verify that you immediately get served a team to scout while in the queue when a match gets created
8. Return to the `/` route, click the `scout` button
9. Verify that you immediately get served a team when joining the queue while there are available robots in the robot-queue
